NorthstarDedicatedTest/keyvalues.cpp

```diff
@@ -88,6 +88,63 @@ void ModManager::TryBuildKeyValues(const char* filename)
 	// load original file, so we can parse out the name of the root obj (e.g. WeaponData for weapons)
 	std::string originalFile = ReadVPKOriginalFile(filename);
 
+	// check if the file was overriden
+	if (m_modFiles.find(normalisedPath) != m_modFiles.end())
+	{
+		ModOverrideFile f = m_modFiles[normalisedPath];
+		// check if any mod owns the file - if it does, this is the first time we're overriding this file.
+		if (f.owningMod != nullptr)
+		{
+			std::ifstream filestream(f.owningMod->ModDirectory / "mod" / normalisedPath);
+			std::stringstream stringstream;
+
+			if (filestream.fail())
+			{
+				spdlog::warn("kv {} was overriden by a mod but no file was found!", ogFilePath);
+				return;
+			}
+			while (filestream.peek() != EOF)
```

Comment:
This, should seriously be a utility function. I'm seeing it get used like 4 times over this project.